django_mongodb/query.py

```diff
@@ -85,15 +85,25 @@ def delete(self):
     def get_cursor(self):
         if self.query.low_mark == self.query.high_mark:
             return []
-        fields = [col.target.column for col in self.columns] if self.columns else None
-        cursor = self.collection.find(self.mongo_query, fields)
+        fields = {}
+        for name, expr in self.columns or []:
+            try:
+                fields[expr.target.column] = 1
+            except AttributeError:
+                # Generate the MQL for an annotation.
+                fields[name] = expr.as_mql_agg(self.compiler, self.connection)
+        pipeline = []
+        if self.mongo_query:
+            pipeline.append({"$match": self.mongo_query})
+        if fields:
+            pipeline.append({"$project": fields})
         if self.ordering:
-            cursor.sort(self.ordering)
+            pipeline.append({"$sort": dict(self.ordering)})
         if self.query.low_mark > 0:
-            cursor.skip(self.query.low_mark)
+            pipeline.append({"$skip": self.query.low_mark})
         if self.query.high_mark is not None:
-            cursor.limit(int(self.query.high_mark - self.query.low_mark))
-        return cursor
+            pipeline.append({"$limit": self.query.high_mark - self.query.low_mark})
+        return self.collection.aggregate(pipeline)
```

Comment:
I think I saw this in other PR. Is there any order to follow?

Reply:
I just rebased this on that PR (you can see it's the first commit here) to make sure everything works with that change. I'll merge that PR first, then rebase this again so this change disappears.